node_buffer.hpp

```diff
@@ -75,41 +76,43 @@ struct node_buffer
 
 	int size() const { return m_current_max_size; }
 
-	// TODO: try to avoid heap allocations here
-	std::string get_nodes(int const num_nodes)
+	// returns two spans containing the peer IP, port and node IDs. The second
+	// span may be empty.
+	std::array<span<char const>, 2>
+	get_nodes(int const num_nodes)
 	{
-		std::string ret;
-
 		if (m_buffer.size() < num_nodes)
```

Comment:
This comparison should be `<=`. No point in returning two ranges if they're going to comprise the entire buffer.